casa/IO/test/tByteSinkSource.cc

```diff
@@ -42,16 +42,14 @@ int main ()
 				     ByteIO::New);
     }
     {
-	RegularFileIO regularFileIO (Path("tByteSinkSource_tmp.dat1"),
-				     ByteIO::New);
-	CanonicalIO canonicalIO(&regularFileIO);
-	CanonicalIO canonicalIO1(canonicalIO);
-	canonicalIO1 = canonicalIO;
+        std::shared_ptr<ByteIO> regularFileIO
+          (new RegularFileIO (Path("tByteSinkSource_tmp.dat1"), ByteIO::New));
+        std::shared_ptr<TypeIO> canonicalIO(new CanonicalIO(regularFileIO));
```

Comment:
```suggestion
        std::shared_ptr<RegularFileIO> regularFileIO
          (new RegularFileIO (Path("tByteSinkSource_tmp.dat1"), ByteIO::New));
        std::shared_ptr<CanonicalIO> canonicalIO(new CanonicalIO(regularFileIO));
```
Additionally, you could use `std::make_shared` here.